### PR TITLE
Add Scheduled snapshots for CC

### PR DESCRIPTION
**Pull request by rlapin-pl**
Add Cloud Composer Scheduled Snapshots support to GA.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
composer: added `recovery_config` in `google_composer_environment` resource
```
Fixes https://github.com/hashicorp/terraform-provider-google/issues/13360

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb
Comment (kubasieron) [inline mmv1/third_party/terraform/resources/resource_composer_environment.go.erb:336]: add AtLeastOneOf logic here

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb
Comment (kubasieron) [inline mmv1/third_party/terraform/resources/resource_composer_environment.go.erb:339]: can be omitted

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb
Comment (kubasieron) [inline mmv1/third_party/terraform/resources/resource_composer_environment.go.erb:346]: strange indentation

File: mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb
Comment (kubasieron) [inline mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb:1791]: maybe us-central1?

File: mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb
Comment (kubasieron) [inline mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb:1807]: strange indentation in this section

File: mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb
Comment (kubasieron) [inline mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb:1814]: this can be skipped

File: mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb
Comment (kubasieron) [inline mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb:1815]: this can be skipped

File: mmv1/third_party/terraform/website/docs/r/composer_environment.html.markdown
Comment (kubasieron) [inline mmv1/third_party/terraform/website/docs/r/composer_environment.html.markdown:286]: add "documented below" link

**Comment by zli82016** [comment]
Added @shuyama1  as the secondary reviewer.

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb
Comment (rlapin-pl) [reply mmv1/third_party/terraform/resources/resource_composer_environment.go.erb:346]: Changed to Optional: true according to the documentation.

File: mmv1/third_party/terraform/website/docs/r/composer_environment.html.markdown
Comment (rlapin-pl) [reply mmv1/third_party/terraform/website/docs/r/composer_environment.html.markdown]: Made it optional

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb
Comment (rlapin-pl) [reply mmv1/third_party/terraform/resources/resource_composer_environment.go.erb]: Done

**Review by zli82016 — APPROVED** [review]
LGTM. Please wait for the review from @shuyama1

**Review by zli82016 — APPROVED** [review]
LGTM. There are still some weird indentations.

File: mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb
Comment (zli82016) [inline mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb:1929]: weird indentations

**Comment by zli82016** [comment]
Please wait for the review from @shuyama1

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb
Comment (shuyama1) [inline mmv1/third_party/terraform/resources/resource_composer_environment.go.erb:374]: Do we want to make one of the fields required or have `AtLeastOneOf` validation for them to prevent users from send empty block `scheduled_snapshots_config{}`?

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb
Comment (shuyama1) [inline mmv1/third_party/terraform/resources/resource_composer_environment.go.erb:356]: Does this field support in-place update?

File: mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb
Comment (shuyama1) [reply mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb:1929]: Yeah, the indentation is bit off here. We use spaces instead of tabs for resource configuration. Would you mind cleaning it up here? Thanks!

**Review by shuyama1 — COMMENTED** [review]
Mostly LGTM! Only some small comments + questions! Thanks!

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb
Comment (rlapin-pl) [reply mmv1/third_party/terraform/resources/resource_composer_environment.go.erb:374]: sending an empty block will be treated as a disabling scheduled snapshots: {enabled: false}

File: mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb
Comment (rlapin-pl) [reply mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb]: Fixed

File: mmv1/third_party/terraform/resources/resource_composer_environment.go.erb
Comment (rlapin-pl) [reply mmv1/third_party/terraform/resources/resource_composer_environment.go.erb:356]: Yes it supports in-place update.

File: mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb
Comment (rlapin-pl) [reply mmv1/third_party/terraform/tests/resource_composer_environment_test.go.erb]: Fixed

**Review by zli82016 — APPROVED** [review]
LGTM. The failed test looks unrelated to this PR. TestAccCloudfunctions2function_cloudfunctions2BasicGcsExample

**Comment by zli82016** [comment]
Please wait for the review from @shuyama1

**Review by shuyama1 — APPROVED** [review]
LGTM! Thanks!
(I removed the GA part in PR title and release-note since this feature is added to both GA and Beta provider)

**Review by zli82016 — APPROVED** [review]
LGTM. You can trigger rechecks by running /gcbrun

**Comment by rlapin-pl** [comment]
@shuyama1 , @zli82016  could you please merge this PR.

**Comment by shuyama1** [comment]
@rlapin-pl Thanks for the ping! Merging it now!

**Pull request merged**
